test/test_sendrecv.jl

```diff
@@ -106,4 +106,18 @@ rreq = nothing
 sreq = nothing
 gc()
 
+root = 0
+g = x -> x^2 + 2x - 1
+if MPI.Comm_rank(comm) == root + 1
```

Comment:
If this test is run on a single process, then `rank+1` is an invalid process. A better way is to use the `dst` and `src` variables defined earlier.

Also, since `src` and `dst` can be equal, you need to use `if` instead of `elseif.